.github/workflows/label-pull-requests.yml

```diff
@@ -0,0 +1,68 @@
+name: Pull request labeler
+
+on:
+  pull_request:
+    types: [opened, reopened, converted_to_draft, ready_for_review, synchronize, edited, review_requested]
+  pull_request_review:
+    types: [submitted, dismissed]
+
+jobs:
+  label_pull_request:
+    runs-on: ubuntu-22.04
+    if: always()
+    permissions:
+      pull-requests: write
+    
+    steps:
+      - name: Label pull request
+        uses: actions/github-script@v6
+        with:
+          script: |
+            function labelsForGenericPullRequestChange() {
+                return ["pr-needs-review"];
+            }
+
+            function labelsForPullRequestReviewSubmitted({ review }) {
+                if (review.state.toLowerCase() === "approved") return ["pr-approved-but-awaiting-ci"];
+
+                return ["pr-waiting-for-author"];
+            }
+
+            function labelsForPullRequestConvertedToDraft() {
+                return [];
+            }
```

Comment:
Oh no you did a thing, change it immediately!
And then reply to this comment.

Reply:
oh no, fixed

Reply:
anotha one

Reply:
comment